ovos_core/intent_services/padatious_service.py

```diff
@@ -314,15 +314,35 @@ def calc_intent(self, utterances: List[str], lang: str = None) -> Optional[Padat
         lang = lang or self.lang
         lang = lang.lower()
         if lang in self.containers:
+            intents = []
             intent_container = self.containers.get(lang)
+
             if self.threaded_inference:
-                with Pool() as pool:
-                    intents = (intent for intent in
-                               pool.imap_unordered(_calc_padatious_intent,
-                                                   ((utt, intent_container)
-                                                    for utt in utterances)))
-                    pool.close()
-                    pool.join()
+                # differences between ThreadPoolExecutor and ProcessPoolExecutor
+                # ThreadPoolExecutor
+                #     Uses Threads, not processes.
+                #     Lightweight workers, not heavyweight workers.
+                #     Shared Memory, not inter-process communication.
+                #     Subject to the GIL, not parallel execution.
+                #     Suited to IO-bound Tasks, not CPU-bound tasks.
+                #     Create 10s to 1,000s Workers, not really constrained.
+                #
+                # ProcessPoolExecutor
+                #     Uses Processes, not threads.
+                #     Heavyweight Workers, not lightweight workers.
+                #     Inter-Process Communication, not shared memory.
+                #     Not Subject to the GIL, not constrained to sequential execution.
+                #     Suited to CPU-bound Tasks, probably not IO-bound tasks.
+                #     Create 10s of Workers, not 100s or 1,000s of tasks.
+
+                self.workers = 4  # do the work in parallel instead of sequentially
```

Comment:
I know I'm late and this got merged but maybe it could detected the number of cores available minus -2 or maybe it could be an option in `mycroft.conf`?

Reply:
done https://github.com/OpenVoiceOS/ovos-core/pull/335